omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/SessionController1_9.java

```diff
@@ -119,4 +129,32 @@ public void delete() {
 		Context.logout();
 	}
 
+	/**
+	 * Get current provider
+	 *
+	 * @return Provider if the user is authenticated
+	 */
+	private Provider getCurrentProvider() {
+		UserContext userContext = Context.getUserContext();
```

Comment:
We do not recommend direct access to the `UserContext`.
Simply use `Context.getAuthenticatedUser()`

Reply:
Ooh! didn't know that. Why is it preferred though?

Reply:
UserContext was meant to be an internal implementation detail not exposed as part of the public API.